src/Core/ISolutionInfoProvider.cs

```diff
@@ -31,6 +31,18 @@ namespace SonarLint.VisualStudio.Core
     /// </remarks>
     public interface ISolutionInfoProvider
     {
+        /// <summary>
+        /// Returns the full directory path for the current solution file, or null if there is not a current solution
+        /// </summary>
+        Task<string> GetSolutionDirectoryAsync();
```

Comment:
There are several classes that need to know the solution directory, rather than the full solution file path.

I considered a number of options:
1. change each of those places to use `ISolutionInfoProvider` and then call `Path.GetDirectory`;
2. add an extension method to the interface that did the same thing
3. add the methods to the interface itself.

I went for option (3) - no duplication in the calling code, and it's immediately obvious that the caller wants the directory.

I would have had to write the same amount of code for (2), and it would have added an unnecessary layer of complexity.